bindings/python/tests/test_legged_odometry.py

```diff
@@ -0,0 +1,93 @@
+                                                               "r_elbow",
+                                                               "l_hip_pitch", "l_hip_roll", "l_hip_yaw",
+                                                               "l_knee", "l_ankle_pitch", "l_ankle_roll",
+                                                               "r_hip_pitch", "r_hip_roll", "r_hip_yaw",
+                                                               "r_knee", "r_ankle_pitch", "r_ankle_roll"])
+    kindyn_desc = blf.construct_kindyncomputations_descriptor(kindyn_handler)
+    assert(kindyn_desc.is_valid())
+
+    dt = 0.01
+    # create configuration parameters handler for legged odometry
+    lo_params_handler = blf.StdParametersHandler()
+    lo_params_handler.set_parameter_float("sampling_period_in_s", dt)
+
+    model_info_group = blf.StdParametersHandler()
+    model_info_group.set_parameter_string("base_link", "root_link")
+    # use same frame for the IMU - requirement due to the software design
+    model_info_group.set_parameter_string("base_link_imu", "root_link")
+    # unused parameters but required for configuration (will be deprecated soon)
+    model_info_group.set_parameter_string("left_foot_contact_frame", "l_foot")
+    model_info_group.set_parameter_string("right_foot_contact_frame", "r_foot")
+    assert(lo_params_handler.set_group("ModelInfo", model_info_group))
+
+    lo_group = blf.StdParametersHandler()
+    lo_group.set_parameter_string("initial_fixed_frame", "l_sole")
+    lo_group.set_parameter_string("initial_ref_frame_for_world", "l_sole")
+    lo_group.set_parameter_vector_float("initial_world_orientation_in_ref_frame",  [1.0, 0.0, 0.0, 0.0])
+    lo_group.set_parameter_vector_float("initial_world_position_in_ref_frame",  [0.0, 0.0, 0.0])
+    lo_group.set_parameter_string("switching_pattern",  "useExternal")
+    assert(lo_params_handler.set_group("LeggedOdom", lo_group))
+
+    # instantiate legged odometry
+    legged_odom = blf.LeggedOdometry()
+    empty_handler = blf.StdParametersHandler()
+    # assert passing an empty parameter handler to false
+    assert(legged_odom.initialize(empty_handler, kindyn_desc.kindyn) == False)
+
+    # assert passing an properly configured parameter handler to true
+    assert(legged_odom.initialize(lo_params_handler, kindyn_desc.kindyn) == True)
+
+    # shape of the robot for the above specified joints list
+    encoders = [-0.0001, 0.0000, 0.0000,
+                0.1570, 0.0003, -0.0000,
+                -0.0609, 0.4350, 0.1833,
+                0.5375,
+                -0.0609, 0.4349, 0.1834,
+                0.5375,
+                0.0895, 0.0090, -0.0027,
+               -0.5694, -0.3771, -0.0211,
+                0.0896, 0.0090, -0.0027,
+               -0.5695, -0.3771, -0.0211]
+    encoder_speeds = [0] * len(encoders)
+
+    time = 0
+    for iter in range(10):
```

Comment:
More idiomatic:

```py
for time in np.arange(start=0, step=dt, stop=10*dt)
```

Reply:
Thanks! Changed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/218/commits/69f96dd14d124f1d152ca304bf8546b1a8066710